[crawler/mobile] crawl sometimes fails due to server errors

Sometimes the Twitter Servers return `503` (Service Unavailable) errors.
The Mobile Crawler then fails:
`[crawler][ERROR] FAILED MCRWL Tommen Baratheon { status: 503, data: '' }`
I'm not sure if the Mobile Search site or the REST API returns this errors. The Crawler should retry after a few seconds in such a case.

https://github.com/Rostlab/JS16_ProjectD_Group4/pull/45/commits/b676261cab14940b87a60778f8f9d9ef3b96daea#diff-726c97f0fff6e3231157be02c25032f2R66 in #45 already contains some code which should handle errors returned by the API but the crawler still fails sometimes.

--- a/crawler/mobile.js
+++ b/crawler/mobile.js
@@ -1,38 +1,39 @@
 const cfg     = require('../core/config'),
       debug   = require('../core/debug')('crawler/mobile', true),
-      https   = require('https'),
+      request = require('request'),
       twitter = require('../crawler/twitter');
 
 var mobile = module.exports = {};
 
 // get returns a Promise for the content of the given URL
 // HTTPS only.
-function get(url) {
+function get(url, retries) {
     return new Promise(function(resolve, reject) {
-        var req = https.get(url, function(res) {
-            var data = '';
-
-            res.on('data', function(chunk) {
-                data += chunk;
-            });
-
-            res.on('end', function() {
-                if (res.statusCode === 200) {
-                    resolve(data);
+        request({
+            url: url,
+            timeout: cfg.twitter.timeout
+        }, function (err, resp, body) {
+            if (!!err) {
+                debug.error(url, err);
+                reject(err);
+            } else if (resp.statusCode === 200) {
+                resolve(body);
+            } else {
+                retries = (!retries) ? 0 : retries;
+                if (resp.statusCode >= 500 && retries < 10) {
+                    debug.warn("Retry because of Server Error", resp.statusCode);
+                    ++retries;
+                    setTimeout(function() {
+                        get(url, retries).then(resolve, reject);
+                    }, (retries*retries)*1000);
                 } else {
                     reject({
-                        status: res.statusCode,
-                        data:   data
+                        status: resp.statusCode,
+                        body:   body
                     });
                 }
-            });
+            }
         });
-
-        req.on('error', function(err) {
-            reject(err);
-        });
-
-        req.end();
     });
 }
 

--- a/package.json
+++ b/package.json
@@ -32,6 +32,7 @@
     "express": "^4.13.4",
     "express-handlebars": "^3.0.0",
     "mongoose": "^4.4.6",
+    "request": "^2.69.0",
     "retext": "^2.0.0",
     "retext-sentiment": "^1.0.0",
     "slug": "^0.9.1",

